lib/pbench/server/api/resources/__init__.py

```diff
@@ -1714,8 +1716,10 @@ def _dispatch(
                     )
             abort(e.http_status, message=str(e), **e.kwargs)
         except Exception as e:
+            x = APIInternalError("Unexpected exception")
```

Comment:
Not that you changed this code to do this in this PR, but shouldn't this try block be:

```
try:
    response = execute(...)
except ...
except ...
else:
    if auditing["finalize"]:
        Audit.create(...)
    return response
```

Any exception in the `Audit` class would cause the API to report an internal error, but the API executed just fine, and then you'd be repeating the "finalize" again.